### PR TITLE
Error on invalid encrypted credentials

### PR DESCRIPTION
**Pull request by sammachin**
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Proposed changes
Fix for https://github.com/node-red/node-red/issues/3496
Also warn if using unencypted credentials 

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red/blob/master/CONTRIBUTING.md)
- [ ] For non-bugfix PRs, I have discussed this change on the forum/slack team.
- [x] I have run `grunt` to verify the unit tests pass
- [x] I have added suitable unit tests to cover the new/changed functionality

### CODE REVIEW

**Comment by sammachin** [comment]
Still working on why the tests fail,

**Comment by Steve-Mcl** [comment]
>  fails

I think the runtime is empty Sam

https://github.com/node-red/node-red/blob/c7f48a83c0a5fa5f1dcd57415822b81899e98b3f/test/unit/%40node-red/runtime/lib/nodes/credentials_spec.js#L453

File: packages/node_modules/@node-red/runtime/locales/en-US/runtime.json [hunk 1]
```diff
@@ -105,7 +105,8 @@
             "error-saving":"Error saving credentials: __message__",
             "not-registered": "Credential type '__type__' is not registered",
             "system-key-warning": "\n\n---------------------------------------------------------------------\nYour flow credentials file is encrypted using a system-generated key.\n\nIf the system-generated key is lost for any reason, your credentials\nfile will not be recoverable, you will have to delete it and re-enter\nyour credentials.\n\nYou should set your own key using the 'credentialSecret' option in\nyour settings file. Node-RED will then re-encrypt your credentials\nfile using your chosen key the next time you deploy a change.\n---------------------------------------------------------------------\n",
-            "unencrypted" : "Using Unecrypted credentials"
+            "unencrypted" : "Using unecrypted credentials",
+            "encryptednotfound" : "Encrypted crededntials not found"
```
Comment (Steve-Mcl) [inline packages/node_modules/@node-red/runtime/locales/en-US/runtime.json:109]: typo "crededntials"

**Comment by knolleary** [comment]
The tests fail in `test/unit/@node-red/runtime/lib/nodes/index_spec.js` because it is hitting this code path, but the mocked runtime object does not include the `log._` function.

The fix is to modify:
https://github.com/node-red/node-red/blob/702545e0b238f160cce9b339808b230f9d741b08/test/unit/%40node-red/runtime/lib/nodes/index_spec.js#L66

to add a placeholder `_` function to the `log` object:

```
        log: {debug:function() {}, warn:function() {}, _: function() {}},
```

File: packages/node_modules/@node-red/runtime/lib/nodes/credentials.js [hunk 2]
```diff
@@ -239,7 +239,15 @@ var api = module.exports = {
                     throw error;
                 }
             } else {
-                credentialCache = credentials;
+                if (encryptionEnabled) {
+                    // Our config expects the credentials to be encrypted but the encrypted object is not found
+                    log.warn(log._("nodes.credentials.encryptednotfound"))
```
Comment (knolleary) [inline packages/node_modules/@node-red/runtime/lib/nodes/credentials.js:244]: Either kebab-case or camelCase - never alllowercase
```suggestion
                    log.warn(log._("nodes.credentials.encryptedNotFound"))
```

File: packages/node_modules/@node-red/runtime/locales/en-US/runtime.json [hunk 3]
```diff
@@ -104,7 +104,9 @@
...
+            "system-key-warning": "\n\n---------------------------------------------------------------------\nYour flow credentials file is encrypted using a system-generated key.\n\nIf the system-generated key is lost for any reason, your credentials\nfile will not be recoverable, you will have to delete it and re-enter\nyour credentials.\n\nYou should set your own key using the 'credentialSecret' option in\nyour settings file. Node-RED will then re-encrypt your credentials\nfile using your chosen key the next time you deploy a change.\n---------------------------------------------------------------------\n",
+            "unencrypted" : "Using unecrypted credentials",
+            "encryptednotfound" : "Encrypted credentials not found"
```
Comment (knolleary) [inline packages/node_modules/@node-red/runtime/locales/en-US/runtime.json:109]: Either kebab-case or camelCase - never alllowercase
```suggestion
            "encryptedNotFound" : "Encrypted credentials not found"
```

File: packages/node_modules/@node-red/runtime/locales/en-US/runtime.json [hunk 4]
```diff
@@ -104,7 +104,9 @@
...
+            "system-key-warning": "\n\n---------------------------------------------------------------------\nYour flow credentials file is encrypted using a system-generated key.\n\nIf the system-generated key is lost for any reason, your credentials\nfile will not be recoverable, you will have to delete it and re-enter\nyour credentials.\n\nYou should set your own key using the 'credentialSecret' option in\nyour settings file. Node-RED will then re-encrypt your credentials\nfile using your chosen key the next time you deploy a change.\n---------------------------------------------------------------------\n",
+            "unencrypted" : "Using unecrypted credentials",
```
Comment (knolleary) [inline packages/node_modules/@node-red/runtime/locales/en-US/runtime.json:108]:
```suggestion
            "unencrypted" : "Using unencrypted credentials",
```

**Review by knolleary — CHANGES_REQUESTED** [review]

**Comment by sammachin** [comment]
Thanks Nick, `npm test` now passes on my machine

**Pull request merged**